components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java

```diff
@@ -260,9 +261,13 @@ public KeyManagerConfiguration getKeyManagerConfiguration() throws APIManagement
     @Override
     public OAuthApplicationInfo mapOAuthApplication(OAuthAppRequest oAuthAppRequest) throws APIManagementException {
         String consumerKey = oAuthAppRequest.getOAuthApplicationInfo().getClientId();
-
+        OAuthApplicationInfo clientInfo = null;
         if (StringUtils.isNotBlank(consumerKey)) {
-            OAuthApplicationInfo clientInfo = retrieveApplication(consumerKey);
+            try{
+            clientInfo = this.getOAuthApplicationInfo(appClient.getApplicationByAppId(consumerKey));
```

Comment:
If the result returned from `appClient.getApplicationByAppId(consumerKey)` is null, there will be a null pointer exception thrown from `this.getOAuthApplicationInfo` method. Let's fix it to first check for null and invoke `getOAuthApplicationInfo()`.